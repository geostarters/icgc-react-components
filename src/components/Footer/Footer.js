// @flow

import React from "react";
import PropTypes from "prop-types";
import { Image } from "semantic-ui-react";

import ResizeComponent from "../../hoc/ResizeComponent/ResizeComponent";

import styles from "./Footer.module.css";

const Footer = ({title = "Nou prototip", customStyleBar = {}, pathLogo = null, width, showRightElements, children, logoSize = "small", customStyleLogo}) => {

	const isMobile = width <= 500;

	return (
		<div className={styles.containerFooter} style={customStyleBar}>

			{ !isMobile && pathLogo &&
				<div className={styles.containerlogo} style={customStyleLogo}>
					<Image src={pathLogo} size={logoSize}/>
				</div>
			}

			<div className={styles.containertitle}>
				{title}
			</div>

			{showRightElements &&
				<div className={styles.containeritemRight}>
					{children}
				</div>
			}

		</div>
	);

};

// eslint-disable-next-line new-cap
export default React.memo(ResizeComponent(Footer));

Footer.propTypes = {
	title: PropTypes.string,
	pathLogo: PropTypes.string,
	logoSize: PropTypes.string,
	showModalInfo: PropTypes.bool,
	width: PropTypes.number,
	customStyleBar: PropTypes.object,
	customStyleLogo: PropTypes.object,
	children: PropTypes.object,
	showRightElements: PropTypes.bool
};