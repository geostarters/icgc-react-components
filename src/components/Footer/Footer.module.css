.containerFooter{
    padding: 1rem;
    background-color: #f5f5f5;
    bottom: 0;
    left: 0;
    width: 100%;
    border-top: 2px solid #ccc;
    display: flex;
	flex-direction: row;
	justify-content: space-between;
    align-items: center;
    flex-shrink: 0;
    position: fixed;
}

.containerlogo {
	max-width: 10em;
	padding: 0 0.5rem;
}

.containertitle{
    font-size: 0.788em;
    padding: 0 0.5rem;
}

.containeritemRight{
    font-size: 0.788em;
    padding: 0 0.5rem;
}
