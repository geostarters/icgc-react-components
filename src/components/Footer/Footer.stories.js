//@flow

import React from "react";

import Footer from "./Footer";

import "semantic-ui-css/semantic.min.css";

export default {
	title: "Footer",
	component: Footer
};

const Template = args => (<Footer {...args}/>);

export const _Footer = Template.bind({});
_Footer.storyName = "Footer";

export const FooterCustomStyle = Template.bind({});
FooterCustomStyle.storyName = "Footer custom style";
FooterCustomStyle.args = {
	title:"Styled",
	customStyleBar: {backgroundColor: "#ccc", color: "#999", position: "relative"}
};

export const FooterLogo = Template.bind({});
FooterLogo.storyName = "Footer logo";
FooterLogo.args = {
	title:"VERSIÓ BETA",
	pathLogo: "https://bolollo.github.io/MapaRegEscPNMM/logo_generalitat_gris.dca7d39f.png",
	logoSize: "small",
	customStyleBar: {backgroundColor: "#ccc", color: "#999"},
	customStyleLogo: {maxWidth: "7em"}
};

export const Footer2Columns = args => (<Footer {...args}>
	<div>
		Test footer
	</div>
</Footer>);
Footer2Columns.storyName = "Footer 2 columns";
Footer2Columns.args = {
	title:"VERSIÓ BETA",
	showRightElements: true
};

export const FooterLogo2Columns = args => (<Footer {...args}>
	<div>
		Test footer
	</div>
</Footer>);
FooterLogo2Columns.storyName = "Footer logo 2 columns";
FooterLogo2Columns.args = {
	title:"VERSIÓ BETA",
	pathLogo: "https://bolollo.github.io/MapaRegEscPNMM/logo_generalitat_gris.dca7d39f.png",
	logoSize: "small",
	customStyleBar: {backgroundColor: "#ccc", color: "#999"},
	customStyleLogo: {maxWidth: "7em"},
	showRightElements: true
};
