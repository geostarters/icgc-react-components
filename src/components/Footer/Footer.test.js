// @flow

import React from "react";
import Footer from "./Footer";
import  { mount } from "enzyme";
import toJSON from "enzyme-to-json";

describe("Footer Component", () => {

	it("renders correct properties", () => {

		const mockProps = {
			title: "Title",
			width: 1000,
			showRightElements: true
		};

		const component = mount(<Footer {...mockProps}></Footer>);

		expect(toJSON(component)).toMatchSnapshot();

	});

	it("renders without image for mobile", () => {

		const mockProps = {
			title: "Title",
			width: 400,
			showRightElements: false,
			pathLogo:"https://bolollo.github.io/MapaRegEscPNMM/logo_generalitat_gris.dca7d39f.png"
		};
		const component = mount(<Footer {...mockProps}></Footer>);

		expect((component.find("img"))).toHaveLength(0);

	});

});
