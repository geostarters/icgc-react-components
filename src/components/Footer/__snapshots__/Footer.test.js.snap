// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component renders correct properties 1`] = `
<ResizeComponent
  showRightElements={true}
  title="Title"
  width={1000}
>
  <Footer
    showRightElements={true}
    title="Title"
    width={1000}
  >
    <footer
      className="containerFooter"
      style={Object {}}
    >
      <div
        className="containertitle"
      >
        Title
      </div>
      <div
        className="containeritemRight"
      />
    </footer>
  </Footer>
</ResizeComponent>
`;
